### PR TITLE
fix(files): do not load legacy files app js

### PR DESCRIPTION
**Pull request by max-nextcloud**
`OCP.Files.Navigation` is undefined.

* Resolves: failures like https://cloud.cypress.io/projects/hx9gqy/runs/11854/overview?roarHideRunsWithDiffGroupsAndTags=1

cypress-screenshot

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests are fixed by this
- [x] Screenshot of test failurs
- [x] Documentation is not required
- [x] Backports are not required

### CODE REVIEW

**Comment by skjnldsv** [comment]
Where is legacy files still loaded?
Those scripts shuold not even be here anymore :)

**Comment by max-nextcloud** [comment]
uh... interesting - i will investigate.

**Comment by max-nextcloud** [comment]
Loaded directly from the html inside the cypress tests. I will see if it also loads without cypress...

grafik

**Comment by max-nextcloud** [comment]
It's served with the initial template when loading `apps/files`:
grafik

**Comment by skjnldsv** [comment]
Ah, found
This ain't supposed to still be there
https://github.com/nextcloud/server/blob/3846ec2899d476c03765859487bce4a8ccd0e799/apps/files/js/app.js#L403

**Comment by skjnldsv** [comment]
It's a multi step, but requires a bit more:
 - Disable [this line](https://github.com/nextcloud/server/blob/be41dc4fa79310328647da48c3426a8eaed75f29/apps/files/lib/Controller/ViewController.php#L221)
 - Remove [systemtagsfilelist](https://github.com/nextcloud/server/blob/0fca8238dd86d572f0f15eaa8e8a5ea836cc76bf/apps/systemtags/src/systemtags.js#L25) 
 - Fix Sidebar to not rely on `OCA.Files.App.fileList.filesClient`
  https://github.com/nextcloud/server/blob/be41dc4fa79310328647da48c3426a8eaed75f29/apps/files/src/services/FileInfo.js#L36-L39

Then it should be good, public pages seems to still work, I gues they load their own minimal stuff

**Comment by max-nextcloud** [comment]
> This ain't supposed to still be there
> 
> https://github.com/nextcloud/server/blob/3846ec2899d476c03765859487bce4a8ccd0e799/apps/files/js/app.js#L403

So can i also drop `initialize()` entirely and `navigation` and `fileList` as they are only written to in `initialize()`?

**Comment by skjnldsv** [comment]
if we don't include the file, dropping the initialize will not matter.
I think we can keep those files here until  https://github.com/nextcloud/server/issues/39914 is complete and we do a cleanup pass

**Comment by max-nextcloud** [comment]
I think I got it... or at least some intermediate state.

Circles is missing the filesList though:
grafik

**Comment by skjnldsv** [comment]
> Circles is missing the filesList though:

Circles is not upgraded for 28 though :thinking: 
I think that's a different topic anyway

**Review by skjnldsv — APPROVED** [review]

File: apps/files/lib/Controller/ViewController.php [hunk 1]
```diff
@@ -218,7 +218,7 @@ public function index($dir = '', $view = '', $fileid = null, $fileNotFound = fal
 
 		// Load the files we need
 		\OCP\Util::addStyle('files', 'merged');
-		\OCP\Util::addScript('files', 'merged-index', 'files');
+		\OCP\Util::addScript('files', 'fileactions');
```
Comment (max-nextcloud) [inline apps/files/lib/Controller/ViewController.php:221]: without the fileactions import files won't open anymore in the viewer. So I guess we still need it.

@skjnldsv - what do you think?

Comment (skjnldsv) [reply apps/files/lib/Controller/ViewController.php:221]: Tried latest Viewer?
We don't use the old `fileactions` ?

Comment (max-nextcloud) [reply apps/files/lib/Controller/ViewController.php:221]: Yes... latest viewer, disabled contacts app which was causing errors and early return from circles app so it does not interfere. Still `.md` `.pdf` and `.jpg`s have no view action and clicking files triggers a download. Did not try other file types.

**Comment by skjnldsv** [comment]
Is it still relevant?

**Comment by max-nextcloud** [comment]
> Is it still relevant?

I think so. Cleaned it up and the main thing now is not to `addScript('files', 'merged-index', 'files');`.

**Review by skjnldsv — APPROVED** [review]

**Review by susnux — APPROVED** [review]

**Pull request merged**